src/InvertedIndices.jl

```diff
@@ -160,4 +164,13 @@ end
 @inline Base.to_indices(A, I::Tuple{Not{<:AbstractArray{Bool, 1}}}) = to_indices(A, (eachindex(IndexLinear(), A),), I)
 @inline Base.to_indices(A, I::Tuple{Not{<:Union{Array{Bool}, BitArray}}}) = to_indices(A, (eachindex(A),), I)
 
+# a cleaner implementation is nt[filter(∉(I.skip), keys(nt))] instead of Base.structdiff, but this would only work on Julia 1.7+
+@inline Base.getindex(nt::NamedTuple, I::InvertedIndex{Symbol}) =
+    getindex(nt, Not((I.skip,)))
+@inline Base.getindex(nt::NamedTuple, I::InvertedIndex{<:AbstractVector{Symbol}}) =
+    getindex(nt, Not(Tuple(I.skip)))
+@inline Base.getindex(nt::NamedTuple, I::InvertedIndex{NTuple{N, Symbol}}) where {N} =
+    I.skip ⊆ keys(nt) ? Base.structdiff(nt, NamedTuple{I.skip}) :
```

Comment:
```suggestion
    allunique(I.skip) && I.skip ⊆ keys(nt) ? Base.structdiff(nt, NamedTuple{I.skip}) :
```
@aplavin + @nalimilan - one last thing to discuss (thank you for your patience and sorry for such a long process but I want to carefully consider all cases).

I propose to add the `allunique` test as this is the rule that normal `NamedTuple` indexing enforces:
```
julia> nt = (a=1,b=2)
(a = 1, b = 2)

julia> nt[(:a, :a)]
ERROR: duplicate field name in NamedTuple: "a" is not unique
```
For vectors uniqueness is not enforced in `Not`:
```
julia> v = 1:3
1:3

julia> v[Not([1,1])]
2-element Vector{Int64}:
 2
 3
```
but AFAICT the reason is that duplicates are allowed with vector indexing:
```
julia> v[[1,1]]
2-element Vector{Int64}:
 1
 1
```
In DataFrames.jl we do not allow duplicates with `Not` on columns:
```
julia> df = DataFrame(rand(3,4), :auto);

julia> df[:, Not([:a, :a])]
ERROR: ArgumentError: Elements of [:a, :a] must be unique
```
(this is not a definitive reference, but it is easier to later allow things as turning error to non-error is non-breaking)

If we agree to this tests would need to be updated.

Thank you!

Reply:
ah - and if we agree the error message would need to be updated.

Reply:
Makes sense, but the error message should probably be different, otherwise it would be misleading. I also hope the performance impact isn't too large.

Reply:
All these extra checks make the functionality less and less useful due to significantly worse performance:
```
julia> nt = (a=1, b=2., c=:def)

# baseline
julia> @btime Base.structdiff($nt, NamedTuple{(:a, :b)})
  1.587 ns (0 allocations: 0 bytes)

# without any checks: my original PR
julia> @btime $nt[Not((:a, :b))]
  1.540 ns (0 allocations: 0 bytes)

# current: subset check
julia> @btime $nt[Not((:a, :b))]
  3.273 ns (0 allocations: 0 bytes)

# with allunique()
julia> @btime $nt[Not((:a, :b))]
  103.062 ns (4 allocations: 384 bytes)
```
I can hardly see myself using this `Not` indexing with two orders of magnitude overhead...

Reply:
Indeed even the constructor in some cases drops this check:
```
julia> (;:a=>1,:a=>2)
(a = 2,)
```
(which is unfortunate I but probably unavoidable)